autoblocks/_impl/testing/run.py

```diff
@@ -275,26 +279,38 @@ async def async_run_test_suite(
     await global_state.http_client().post(f"{cli()}/end", json=dict(testExternalId=test_id))
 
 
+# Sync fn
+@overload
 def run_test_suite(
     id: str,
     test_cases: Sequence[TestCaseType],
     evaluators: Sequence[BaseTestEvaluator[TestCaseType, OutputType]],
-    fn: Callable[[TestCaseType], Union[OutputType, OutputType]],
+    fn: Callable[[TestCaseType], OutputType],
+    max_test_case_concurrency: int = DEFAULT_MAX_TEST_CASE_CONCURRENCY,
+) -> None: ...
+
+
+# Async fn
+@overload
+def run_test_suite(
+    id: str,
+    test_cases: Sequence[TestCaseType],
+    evaluators: Sequence[BaseTestEvaluator[TestCaseType, OutputType]],
+    fn: Callable[[TestCaseType], Awaitable[OutputType]],
+    max_test_case_concurrency: int = DEFAULT_MAX_TEST_CASE_CONCURRENCY,
+) -> None: ...
+
+
+def run_test_suite(
+    id: str,
+    test_cases: Sequence[TestCaseType],
+    evaluators: Sequence[BaseTestEvaluator[TestCaseType, OutputType]],
+    fn: Union[Callable[[TestCaseType], OutputType], Callable[[TestCaseType], Awaitable[OutputType]]],
```

Comment:
ideally we wouldn't need the overloads and this type would work out of the box but ran into https://github.com/python/mypy/issues/16646